multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/ProcessContext.java

```diff
@@ -36,14 +40,18 @@ public CloudControllerClient getControllerClient() {
         String spaceGuid = getVariable(Variables.SPACE_GUID);
         String correlationId = getVariable(Variables.CORRELATION_ID);
         CloudControllerClient delegate = clientProvider.getControllerClient(userGuid, spaceGuid, correlationId);
-        return new LoggingCloudControllerClient(delegate, stepLogger);
+        Set<String> secretParameters = VariableHandling.get(execution, Variables.SECURE_EXTENSION_DESCRIPTOR_PARAMETER_NAMES);
+        DynamicSecureSerialization dynamicSecureSerialization = SecureSerializationFactory.ofAdditionalValues(secretParameters);
```

Comment:
These two lines follow the same pattern and appear in several files (with minor variations). Might be worth extracting into a small utility or helper.